"Cannot find module" after upgrading from 0.8.1 to 1.0.1

Getting the following exception after upgrading:

```
10:12:25 AM [vite] Internal server error: Cannot find module '@/components/Flex.ts'
Require stack:
- /Users/leonidkhachaturov/code/space/darkmatter-web/src/components/Centered/Centered.ts in/Users/leonidkhachaturov/code/space/darkmatter-web/src/components/Centered/Centered.ts

  Plugin: wyw-in-js
  File: /Users/leonidkhachaturov/code/space/darkmatter-web/src/components/Centered/Centered.ts
      at Module.evaluate (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/module.js:268:13)
      at evaluate (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/evaluators/index.js:14:5)
      at BaseAction.evalFile [as handler] (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/generators/evalFile.js:35:43)
      at evalFile.next (<anonymous>)
      at /Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/BaseAction.js:71:78
      at EventEmitter.action (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/utils/EventEmitter.js:25:22)
      at BaseAction.emitAction (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/BaseAction.js:136:39)
      at nextFn (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/BaseAction.js:71:32)
      at processNext (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/BaseAction.js:116:28)
      at Object.next (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/BaseAction.js:125:9)
      at asyncActionRunner (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/actionRunner.js:39:102)
      at asyncActionRunner (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform/actions/actionRunner.js:46:28)
      at async transform (/Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/transform/lib/transform.js:118:20)
      at async TransformPluginContext.transform (file:///Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/@wyw-in-js/vite/esm/index.mjs:372:22)
      at async EnvironmentPluginContainer.transform (file:///Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/vite/dist/node/chunks/node.js:31351:14)
      at async loadAndTransform (file:///Users/leonidkhachaturov/code/space/darkmatter-web/node_modules/vite/dist/node/chunks/node.js:26319:26)
```

`@` is an alias defined in vite.config.js:

```
"@": path.resolve(__dirname, "./src")
```

I have tried both Vite 8 beta and Vite 7.3.1, both don't work.

--- a/.changeset/fix-vite-external-resolve.md
+++ b/.changeset/fix-vite-external-resolve.md
@@ -1,0 +1,5 @@
+---
+'@wyw-in-js/vite': patch
+---
+
+Avoid falling back to Node resolution for Vite `external` resolved file ids (incl. `external: "absolute"`), which could break aliased imports during build-time evaluation (SSR/dev).

--- a/packages/vite/src/__tests__/asyncResolve.test.ts
+++ b/packages/vite/src/__tests__/asyncResolve.test.ts
@@ -212,4 +212,71 @@ describe('vite asyncResolve', () => {
       rmSync(tempDir, { recursive: true, force: true });
     }
   });
+
+  it('does not fall back to syncResolve for external resolved file ids', async () => {
+    const { default: wywInJS } = await import('../index');
+    const plugin = wywInJS();
+
+    plugin.configResolved({
+      root: process.cwd(),
+      mode: 'development',
+      command: 'serve',
+      base: '/',
+    } as any);
+
+    const tempDir = mkdtempSync(join(tmpdir(), 'wyw-vite-fs-external-'));
+    try {
+      const filePath = join(tempDir, 'Flex.ts');
+      writeFileSync(filePath, 'export const Flex = () => null;', 'utf8');
+
+      const viteFsId = `/@fs/${filePath.replace(/\\/g, '/')}`;
+      requestedId = viteFsId;
+
+      const resolveMock = jest.fn().mockResolvedValue({
+        id: viteFsId,
+        external: 'absolute',
+      });
+
+      await plugin.transform?.call(
+        { resolve: resolveMock } as any,
+        'console.log("test")',
+        '/entry.tsx'
+      );
+
+      expect(syncResolveMock).not.toHaveBeenCalled();
+      expect(asyncResolveResults).toContain(normalize(filePath));
+    } finally {
+      rmSync(tempDir, { recursive: true, force: true });
+    }
+  });
+
+  it('falls back to syncResolve for external bare specifiers', async () => {
+    const { default: wywInJS } = await import('../index');
+    const plugin = wywInJS();
+    requestedId = 'react';
+
+    const resolvedPath = join(__dirname, 'node_modules', 'react.js');
+    syncResolveMock.mockReturnValue(resolvedPath);
+
+    plugin.configResolved({
+      root: process.cwd(),
+      mode: 'development',
+      command: 'serve',
+      base: '/',
+    } as any);
+
+    const resolveMock = jest.fn().mockResolvedValue({
+      id: requestedId,
+      external: true,
+    });
+
+    await plugin.transform?.call(
+      { resolve: resolveMock } as any,
+      'console.log("test")',
+      '/entry.tsx'
+    );
+
+    expect(syncResolveMock).toHaveBeenCalledWith('react', '/entry.tsx', []);
+    expect(asyncResolveResults).toContain(resolvedPath);
+  });
 });

--- a/packages/vite/src/index.ts
+++ b/packages/vite/src/index.ts
@@ -355,14 +355,6 @@ export default function wywInJS({
       const log = logger.extend('vite').extend(getFileIdx(importer));
 
       if (resolved) {
-        if (resolved.external) {
-          // If module is marked as external, Rollup will not resolve it,
-          // so we need to resolve it ourselves with default resolver
-          const resolvedId = syncResolve(what, importer, stack);
-          log("resolve ✅ '%s'@'%s -> %O\n%s", what, importer, resolved);
-          return resolvedId;
-        }
-
         log("resolve ✅ '%s'@'%s -> %O\n%s", what, importer, resolved);
 
         // Vite adds param like `?v=667939b3` to cached modules
@@ -380,6 +372,17 @@ export default function wywInJS({
 
         if (resolvedId.startsWith('/@')) {
           return null;
+        }
+
+        if (resolved.external) {
+          // In SSR mode Vite can mark resolved file ids as `external` (including `external: "absolute"`).
+          // Prefer Vite's resolved file path and only fall back to Node resolution for bare specifiers.
+          if (path.isAbsolute(resolvedId) || existsSync(resolvedId)) {
+            return resolvedId;
+          }
+
+          // If module is marked as external, Rollup will not resolve it further, so resolve it ourselves.
+          return syncResolve(what, importer, stack);
         }
 
         if (!existsSync(resolvedId)) {

